src/AppBundle/Form/ProcurationProfileType.php

```diff
@@ -51,10 +64,80 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                     'day' => 'JJ',
                 ],
             ])
+            ->add('state', HiddenType::class, [
+                'mapped' => false,
+            ])
         ;
+
+        $builder->addEventListener(FormEvents::PRE_SUBMIT, [$this, 'onPreSubmit']);
+    }
+
+    public function onPreSubmit(FormEvent $event): void
+    {
+        $request = $event->getData();
+        if (!is_array($request)) {
+            throw new UnexpectedTypeException($request, 'array');
```

Comment:
Please use a `TranformationFailedException` in submit related events so the client gets an error and not a 500.